processing/src/main/java/org/apache/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -134,6 +134,11 @@ public Map<String, ColumnAnalysis> analyze(Segment segment)
             analysis = analyzeStringColumn(capabilities, storageAdapter, columnName);
           }
           break;
+        case ARRAY:
+          // todo (clint): this is wack, but works for now because arrays are always nested complex columns...
```

Comment:
Instead of a todo like this.  How about an if check that validates that it's a nested column and, if it's not a nested column, throws an Exception that is clearly saying that the code is currently written assuming that the arrays are delivered only by nested columns?

Reply:
yep, i'm planning on replacing this todo with a 'what we need to do in the future', just haven't got to it yet

Reply:
split out into an `analyzeArrayColumn` that is a mostly empty stub